pkgs/misc/cups/drivers/fxlinuxprintutil/default.nix

```diff
@@ -0,0 +1,46 @@
+{ stdenv, lib, fetchzip, dpkg, autoPatchelfHook, cups, tcl, tk, xorg, makeWrapper }:
+let
+  debPlatform =
+    if stdenv.hostPlatform.system == "x86_64-linux" then "amd64"
+    else if stdenv.hostPlatform.system == "i686-linux" then "i386"
+         else throw "Unsupported system: ${stdenv.hostPlatform.system}";
+in
+stdenv.mkDerivation rec {
+  name = "fxlinuxprintutil-${version}";
+  version = "1.1.1-1";
+
+  src = fetchzip {
+    url = "https://onlinesupport.fujixerox.com/driver_downloads/fxlinuxpdf112119031.zip";
+    sha256 = "1mv07ch6ysk9bknfmjqsgxb803sj6vfin29s9knaqv17jvgyh0n3";
+    curlOpts = "--user-agent Mozilla/5.0";  # HTTP 410 otherwise
+  };
+
+  patches = [ ./fxlputil.patch ./fxlputil.tcl.patch ./fxlocalechk.tcl.patch ];
+
+  nativeBuildInputs = [ dpkg autoPatchelfHook makeWrapper ];
+  buildInputs = [ cups tcl tk ];
+
+  sourceRoot = ".";
+  unpackCmd = "dpkg-deb -x $curSrc/fxlinuxprintutil_${version}_${debPlatform}.deb .";
+
+  dontConfigure = true;
+  dontBuild = true;
+
+  installPhase = ''
+    mkdir -p $out
+    mv usr/bin $out
+    mv usr/lib $out
+
+    substituteInPlace $out/bin/fxlocalechk.tcl --subst-var-by libX11 "${xorg.libX11}"
```

Comment:
we can do a really nice thing in patches that I personally think is much cleaner
```
patches = [
  (substituteAll { 
    src = ./fxlocalechk.tcl.patch;
    inherit (xorg) libX11;
  })
];

```

Reply:
done in d03c4109d699670f1bfe11794378c82ee7d26499